### ISSUE TITLE
TS warning when running with --typeCheck

### ISSUE THREAD

**Issue by briancavalier** [TylorS/typed-test#10]
I've started using `@typed/test` in [`@most/index`](https://github.com/mostjs/index/pull/3), and just added the `--typeCheck` flag to enable type checking when running tests.  That yielded a type warning that doesn't seem like it should happen:

```
➜ npm test

> @most/index@1.0.0 test /Users/brian/Projects/cujojs/@most/index
> typed-test --typeCheck 'src/**/*.test.ts'

Running tests...
Typechecking...
Typechecking complete.
/Users/brian/Projects/cujojs/@most/index/Users/brian/Projects/cujojs/@most/index/node_modules/@typed/test/lib/results/resultToDom.d.ts (2,62):
  Cannot find name 'HTMLElement'.
/Users/brian/Projects/cujojs/@most/index/Users/brian/Projects/cujojs/@most/index/node_modules/@typed/test/lib/results/resultToDom.d.ts (3,76):
  Cannot find name 'HTMLElement'.
```

The project isn't specifically using any DOM-related.  The path in the error message looks like it's been double-absolute-ized as well.  The tests still run successfully, so it's not a blocker.

Any ideas what's going on?

**Comment by Frikki** [comment]
`@typed/test` is able to run browser tests and uses the `HTMLElement` [here](https://github.com/TylorS/typed-test/blob/c64674c9ccc0903378f619fd904912b1f9e16e0a/source/results/resultToDom.ts) and [here](https://github.com/TylorS/typed-test/blob/c64674c9ccc0903378f619fd904912b1f9e16e0a/source/browser/api/reportResults.ts).

There are two workarounds that I know of:

1. add `"dom"` to the list of libraries in _tsconfig.json_, or
2. split the tsconfig in two as described [here](https://github.com/TylorS/typed-test/issues/9#issuecomment-478987898).

**Comment by TylorS** [comment]
@briancavalier Thanks for bug report - I feel issue is a leaking of @typed/test's requirement for `dom` to be in your `lib` configuration of your tsconfig.json. The fix may be similar to that of #9 where I should auto-add the requirement for `dom` to the parsed compiler options. What do you guys think?

**Comment by briancavalier** [comment]
That seems pretty reasonable, @TylorS.  In my mind, an ideal end result would be:

1. Users of `@typed/test` only specify what their project needs in tsconfig.json.  IOW, they can use the same `tsconfig.json` when they use `@typed/test` as they would if they _weren't_ using it.
2. In that scenario, `@typed/test` does whatever it needs to do (auto-adding options, etc.) to ensure that tests can be run in node and/or browser with no surprises.

**Comment by TylorS** [comment]
Yes - spot on. I really want to avoid situations where `@typed/test` imposes arbitrary configuration changes on you
